pyproject.toml

```diff
@@ -22,42 +22,73 @@ required-version = "22.12.0"
+include = [
+  ".buildkite/dagster-buildkite",
+  "python_modules",
+  "examples",
+  "integration_tests",
+  "scripts",
+]
+
+# Unfortunately pyright does not offer a way to extend the default exclusions, so we have to
+# reiterate them here if we want to add anything else.
+exclude = [
+  "**/node_modules",
+  "**/__pycache__",
+  "**/__generated__",
+  "**/vendor",
+  "**/.tox",
+  ".git",
+  "**/.venv*",
+]
+
+# These two settings point pyright to a python environment to resolve imports against. This virtual
+# environment is defined in the `pyright` tox environment in the tox section below-- that
+# environment must be built before pyright can run correctly.
+venv = ".venv"
+venvPath = "pyright/envs/master"
+
+# Set to false to help us during the transition from mypy to pyright. Mypy does
+# not analyze unannotated functions by default, and so as of 2023-02 the codebase contains a large
+# number of type errors in unannotated functions. Eventually we can turn off this setting.
+analyzeUnannotatedFunctions = false
 
-# Ignore cases where mypy cannot resolve *the types* for a given import. Note that this is different
-# than not being able to resolve the import at runtime. A module can be installed, but mypy will
-# consider it unresolveable if it does not either (a) have a `py.typed` marker file; (b) have
-# a corresponding stub package (i.e. named `types-XXX`) available to mypy. Because Dagster uses
-# has several dependencies that do not satisfy either criterion, mypy will by default emit errors
-# when it encounters imports of those dependencies. Since there's no way to fix these, we silence
-# them.
-ignore_missing_imports = true
+# Minimum version of Python on which code must run. This determines the standard library stubs used by
+# pyright.
+pythonVersion = "3.7"
 
-# If a type stub package is missing from the environment but available on typeshed, automatically
-# install it. Note that without `--non-interactive` this will prompt the user.
-install_types = true
+# Use "basic" ruleset. This differs from strict in several ways, but most
+# importantly it does not flag untyped code as an error.
+typeCheckingMode = "basic"
 
-# Allow PEP-420-style namespace packages. Without this enabled, different parts of a namespace
-# package will trigger "Duplicate module" errors.
-namespace_packages = true
+# Disable reading type annotations from libraries that are not explicitly marked as typed (i.e. that
+# include a py.typed file). All imports from these libraries are given the `Unknown` type (i.e.
+# `Any`). This setting does not affect `py.typed` libraries.
+useLibraryCodeForTypes = false
 
-# Print codes (e.g. "[arg-type]") for each error in output.
-show_error_codes = true
+# We use ruff for this.
+reportInvalidStringEscapeSequence = false
+
+# As of 2023-02-02, there are still many `py.typed` libs that are not compliant with the standards
+# for defining a public API.
+reportPrivateImportUsage = false
```

Comment:
wait so now we are turning this off?

Reply:
In that DM I said we'd turn it off (but that handles only _some_ cases)-- I edited it maybe an hour or so after first sent:

>Given this, the best we can do is turn off reportPrivateImportUsage, which will affect pendulum and datadog but not e.g. sqlalchemy and several other libs that required import changes. (edited)

Reply:
Ah got thanks for clarification